lib/jxl/encode.cc

```diff
@@ -2258,64 +2242,37 @@ JxlEncoderStatus JxlEncoderAddImageFrameInternal(
     return JxlErrorOrStatus::Success();
   }
 
-  size_t row_offset;
-  const void* buffer = input.get_color_channel_data_at(
-      input.opaque, 0, 0, xsize, ysize, &row_offset);
-  if (!buffer) {
-    return JXL_API_ERROR(frame_settings->enc, JXL_ENC_ERR_GENERIC,
-                         "no buffer for color channels given");
+  if (!streaming) {
+    if (!frame_data.CopyBuffers()) {
```

Comment:
I assume that this happens because we need to read the frame data after the point where it is guaranteed to be available anymore - in that case, could you add a comment? (and even not in that case...)

Reply:
Done.